task_2/andrii.lukianchykov/task_2/task_2/task_2/src/main/java/com/bookshop/bookdatabase/BookDataBase.java

```diff
@@ -0,0 +1,85 @@
+package com.bookshop.bookdatabase;
+
+import com.bookshop.book.Book;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.lang.System.out;
+
+public class BookDataBase {
+
+    private Book book;
+    private static List<Book> listOfBooks = new ArrayList<>();
+
+    public BookDataBase() {}
+
+    public BookDataBase(List<Book> listOfBooks) {
+        this.listOfBooks = listOfBooks;
+
+    }
+    public  BookDataBase(Book book) {
+        this.book = book;
+    }
+
+    public void defaultValuesListOfBooks() {
+
+        getListOfBooks().add(new Book("Legend of Searcher", "Terry Goodkind", "2008"));
+        getListOfBooks().add(new Book("Comics", "Overwatch", "2017"));
+        getListOfBooks().add(new Book("Limited Collection of how to code right", "Java Enetrprise", "2017"));
+    }
+
+    public void isEmptyBookOb() {
+
+        if(getBook() != null) {
+
+            getListOfBooks().add(getBook());
+        }
+    }
+
+    public List<Book> findBookbyA(String author) {
+
+        List<Book> books = new ArrayList<Book>();
+
+        for(Book b :  getListOfBooks() ) {
+
+            if (b.getAuthorBook().equals(author)) {
+                books.add(getBook());
+            }
+
+        }
+
+        return books;
+    }
+    public Book getBook() {
+        return book;
+    }
+
+    public void getNewBook(Book book) {
+
+        if(book != null) {
+
+            List<Book> books = new ArrayList<>();
```

Comment:
The method does nothing. The next two lines of code just create books list and then adds a book in it. Finally list will be garbage collected.